### PR TITLE
Add geoBackupPolicies and serverCommunicationLinks swagger and examples

### PR DESCRIPTION
**Pull request by nathannfan**
GeobackupPolicies - Update only singleton resource with GET and PUT operations

serverCommunicationLinks - Create only resource with GET, PUT and DELETE.  Update is not supported on this resource.  I'm open to suggestions on whether CommunicationLinks_Create should be used, or if I should use CommunicationLinks_CreateOrUpdate so that we don't have breaking changes when API is updated.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by nathannfan** [comment]
@dsgouda would y'all be okay with leaving ServerCommunicationLinks PUT as CreateOrUpdate, even though update isn't yet supported?

**Comment by dsgouda** [comment]
@nathannfan Thanks for opening the PR, I shall respond by tomorrow.

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (dsgouda) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: Update must be a patch operation.

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (dsgouda) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: Just making sure that the readonly here is intentional. Marking `readonly: true` indicates this is not a tracked resource.

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (dsgouda) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: nit: make descriptions more meaningful, eg.: "Communication links successfully returned", applicable to all other descriptions too

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (dsgouda) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Same question as above regarding the location property.

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
@nathannfan if Update is not supported please rename it to `_Create`

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: Then we can call it CreateOrUpdate, but technically no creating is happening.

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: Mutability for tracked resources is [create, read] but since this resource can't actually be created, I have it as readOnly.

Do tracked resources always have tags?  This resource didn't have them in fiddler.

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Our general approach has been to just make this HTTP status code name.  Has this changed?

**Comment by jaredmoo** [comment]
I will review tomorrow. Thx

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Do tracked resources always have tags? This resource didn't have them in fiddler.  And location wasn't required for resource create, which I believe it is for tracked resources.

**Comment by jaredmoo** [comment]
@dsgouda it is possible that PUT may support update in the future, so naming it CreateOrUpdate might be wise to future-proof the client SDKs.

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: Since this is singleton, should we go with Databases_CreateOrUpdateGeoBackupPolicy operation id instead?

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: assuming this is a proxy resource, should inherit ProxyResource instead of Resource directly

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: We have several proxy resources that return location as a readonly property. Looks like this is one of those.

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: nit: remove unused?

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Should it be ServerCommunicationLinks_Delete instead?

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: also ProxyResource

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Nit: capitalization of `Server`

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Minor feedback, mostly looks great!

**Comment by nathannfan** [comment]
@jaredmoo @dsgouda  Yeah I should've been more clear that update may be added in the future.  Could we leave it CreateOrUpdate to avoid a breaking change?

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (dsgouda) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Is there a reason for this documentation, per all other operationIds this should be `CommunicationLinks_delete`

Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: I mean that all of these operations should be ServerCommunicationLinks_X

**Comment by jaredmoo** [comment]
The PUT http method currently only does create, but it might do create or update in the future. Hence PUT should be _CreateOrUpdate. Yes, PATCH would be called _Update.

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Confirmed in our manifest that this is a proxy resource w/ location

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: Sounds reasonable

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: Pretty sure this is not a tracked resource, but for some reason it doesn't even show up in our CSM manifest at all.

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Changed

File: arm-sql/2014-04-01/swagger/serverCommunicationLinks.json
Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/serverCommunicationLinks.json]: Changed

**Review by dsgouda — APPROVED** [review]
Looks OK at my end, waiting for @jaredmoo 's approval.

File: arm-sql/2014-04-01/swagger/geoBackupPolicies.json
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/geoBackupPolicies.json]: tpyo `creating`

**Review by jaredmoo — APPROVED** [review]

**Comment by nathannfan** [comment]
@dsgouda Thanks, Deepak!

**Pull request merged**
